burn-dataset/src/dataset/sqlite.rs

```diff
@@ -151,19 +200,190 @@ fn fetch_columns_and_len(
 }
 
 /// Create a connection pool and make sure the connections are read only
-fn create_conn_pool(db_file: &str) -> Pool<SqliteConnectionManager> {
+fn create_conn_pool(db_file: &str, create_db: bool) -> Pool<SqliteConnectionManager> {
+    let sqlite_flags = if create_db {
+        OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE
+    } else {
+        OpenFlags::SQLITE_OPEN_READ_ONLY
+    };
+
     // Create a connection pool and make sure the connections are read only
-    let manager =
-        SqliteConnectionManager::file(db_file).with_flags(OpenFlags::SQLITE_OPEN_READ_ONLY);
+    let manager = SqliteConnectionManager::file(db_file).with_flags(sqlite_flags);
     let conn_pool: Pool<SqliteConnectionManager> = Pool::new(manager).unwrap();
     conn_pool
 }
 
+pub struct SqliteDatasetSaver<I> {
+    name: String,
+    db_file: Option<String>,
+    splits: Option<Vec<String>>,
+    overwrite: bool,
+    base_dir: Option<String>,
+    conn_pool: Option<Pool<SqliteConnectionManager>>,
+    is_initialized: bool,
+    phantom: PhantomData<I>,
+}
+
+impl<I> SqliteDatasetSaver<I>
+where
+    I: Clone + Send + Sync + Serialize + DeserializeOwned,
+{
+    /// Create a new SqliteDatasetSaver.
+    pub fn new(name: &str) -> Self {
+        SqliteDatasetSaver {
+            name: name.to_string(),
+            db_file: None,
+            splits: None,
+            base_dir: None,
+            overwrite: false,
+            conn_pool: None,
+            is_initialized: false,
+            phantom: PhantomData::default(),
+        }
+    }
+
+    /// Set the splits.
+    pub fn with_splits(mut self, splits: &[&str]) -> Self {
+        self.splits = Some(splits.iter().map(|s| s.to_string()).collect());
+        self
+    }
+
+    /// Check if the dataset exists.
+    pub fn exists(&self) -> bool {
+        let db_file = self.db_file();
+        Path::new(&db_file).exists()
+    }
+
+    /// Return the SqliteDataset for the given split.
+    pub fn dataset(&self, split: &str) -> SqliteDataset<I> {
+        if !self.exists() {
+            panic!("Dataset does not exist: {}", self.name);
+        }
+
+        SqliteDataset::from_db_file(self.db_file().as_str(), split, true)
+    }
+
```

Comment:
What is the reason for the saver struct to expose the dataset function? I think the saver should only expose save and the options (with_splits, with_base_dif, etc.), maybe the name is wrong?

Reply:
After a sqlite db is created, we would to get a dataset access. It's a convenience instead of passing a db location and getting a dataset. 

I can rename the method to `get_dataset`, maybe?

Reply:
Per our offline conversation. This method should go away because we do not wish to get a reader while the data is being written.